zerver/lib/types.py

```diff
@@ -221,6 +221,49 @@ class NeverSubscribedStreamDict(TypedDict):
     subscribers: NotRequired[List[int]]
 
 
+class StreamClientDict(TypedDict):
```

Comment:
What do you think about `APIStreamDict` for this object?  I think having "API" in the name might be more obvious in meaning to folks reading the code, and we've used this before (though in a type removed in 072051f81e46b2a933ad9d6b9da73a0145909304).

Reply:
This makes sense. And similarily we can have `APISubscriptionDict`.